js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt

```diff
@@ -49,15 +47,17 @@ class MainActivity : AppCompatActivity() {
         val disableTranslationButton = findViewById<Button>(R.id.disableTranslationButton)
         val disableLanguageDetectionButton = findViewById<Button>(R.id.disableLanguageDetectionButton)
         val setLanguage = findViewById<EditText>(R.id.editTextLanguage);
-        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView() } }
-        disableGrammarButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableGrammar") } }
-        disableTranslationButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableTranslation") } }
-        disableLanguageDetectionButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableLanguageDetection", setLanguage.text.toString()) } }
+        val checkBoxToken = findViewById<CheckBox>(R.id.checkBoxToken);
+        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView(tokenFromServer = checkBoxToken.isChecked) } }
```

Comment:
tokenFromServer = checkBoxToken.isChecked

Should create a local variable for this that you pass to all the following calls, _getTokenFromServer_

Reply:
done